src/microcosm.js

```diff
@@ -82,113 +67,75 @@ inherit(Microcosm, Emitter, {
     this.removeAllListeners()
   },
 
-  /**
-   * Generates the starting state for a Microcosm instance by asking every
-   * domain that subscribes to `getInitialState`.
-   */
   getInitialState () {
-    return this.realm.invoke('getInitialState', {})
+    return this.initial
+  },
+
+  recall (action) {
+    return this.archive.get(action)
+  },
+
+  save (action, state) {
+    return this.archive.set(action, state)
+  },
+
+  clean (action) {
+    this.archive.remove(action)
   },
 
-  /**
-   * Dispatch an action to a list of domains. This is used by state management
-   * methods, like `rollforward` and `getInitialState` to compute state.
-   * Assuming there are no side-effects in domain handlers, this is pure.
-   * Calling this method will not update repo state.
-   */
   dispatch (state, action) {
     let handlers = this.realm.register(action)
     let current = state
 
     for (var i = 0, len = handlers.length; i < len; i++) {
-      var { key, domain, handler, length } = handlers[i]
-
-      var next = null
-
-      switch (length) {
-        case 0:
-          next = handler.call(domain)
-          break;
-        case 1:
-          next = handler.call(domain, get(state, key))
-          break;
-        case 2:
-        default:
-          next = handler.call(domain, get(state, key), action.payload)
-      }
+      var { key, domain, handler } = handlers[i]
+
+      var last = get(state, key)
+      var next = handler.call(domain, last, action.payload)
```

Comment:
This was a perf opt, but it throws off Jest and I don't want to hit that on a project.